pkg/dyninst/compiler/logical/functions.go

```diff
@@ -44,6 +44,7 @@ type ProcessExpression struct {
 }
 
 // Process user data of a specific type, chasing pointers, resolving interfaces, etc (after the data was already read into ringbuf).
+// Output offset must be set to the beginning of the data. The function implementation must advance the offset to the end of the data.
```

Comment:
I think it'd help the casual reader if you'd say "The generated function expects the output offset to be set...", to make it clear that we're talking at a meta level.